test/test_torch.py

```diff
@@ -1973,6 +1973,11 @@ def test_tensor_factory(self):
             a[0] = 7.
             self.assertEqual(5., res1[0].item())
 
+            # test ill-sized strides raise exception
+            b = np.array([3., 5., 8.])
+            b.strides = (3,)
+            self.assertRaises(ValueError, lambda: torch.tensor(b))
```

Comment:
Numpy is not always available, so you have to guard this test. Put this in a separate method, and add `@unittest.skipIf(not TEST_NUMPY, "Numpy not found")`

Reply:
Isn't that what `if TEST_NUMPY:` on L1969 is for? I just added to this conditional.  I can move the whole block out if you prefer.

Reply:
Actually we should put the test right next to existing code that checks `torch.from_numpy`, not `torch.tensor`. The later one copies the code, so theoretically it should handle this case just fine. The first one can't possibly handle it, because it would violate the invariant that the returned tensor shares memory.